.github/workflows/maven-checks.yml

```diff
@@ -46,5 +46,72 @@ jobs:
         run: |
           export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
           ./mvnw install -B -V -T 1C -DskipTests -Dmaven.javadoc.skip=true --no-transfer-progress -P ci -pl '!presto-test-coverage,!:presto-docs'
+      - name: Upload presto-server tarball
+        if: matrix.java == '8.0.442'
+        uses: actions/upload-artifact@v4
+        with:
+          name: presto-server
+          path: presto-server/target/presto-server-0.293.tar.gz
+      - name: Upload presto-cli executable
+        if: matrix.java == '8.0.442'
+        uses: actions/upload-artifact@v4
+        with:
+          name: presto-cli
+          path: presto-cli/target/presto-cli-0.293-executable.jar
       - name: Clean Maven Output
         run: ./mvnw clean -pl '!:presto-server,!:presto-cli,!presto-test-coverage'
+  
+  presto-coordinator-with-clp-connector-runtime-image:
+    name: presto-coordinator-runtime-image
+    needs: maven-checks
+    runs-on: ubuntu-22.04
+    if: ${{ always() && success() }}
+    steps:
+      - uses: "actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683"
+        with:
+          submodules: "recursive"
+
+      - name: "Login to Image Registry"
+        uses: "docker/login-action@74a5d142397b4f367a81961eba4e8cd7edddf772"
+        with:
+          registry: ghcr.io
+          username: ${{github.actor}}
+          password: ${{secrets.GITHUB_TOKEN}}
+      
+      - name: Download presto-server
+        uses: actions/download-artifact@v4
+        with:
+          name: presto-server
+          path: /tmp
+
+      - name: Download presto-cli
+        uses: actions/download-artifact@v4
+        with:
+          name: presto-cli
+          path: /tmp
+
+      - name: "Run build.sh"
+        working-directory: ./docker
+        run: |
+          mv /tmp/presto-server-0.293.tar.gz ./
+          mv /tmp/presto-cli-0.293-executable.jar ./
+  
+          if [ "${{ github.event_name }}" = "push" ] && \
+             [ "${{ github.repository }}" = "y-scope/presto" ] && \
+             [ "${{ github.ref }}" = "refs/heads/release-0.293-clp-connector" ]; then
+            export PUBLISH=true
+          else
+            echo "github.event_name = ${{ github.event_name }}"
+            echo "github.repository = ${{ github.repository }}"
+            echo "github.ref = ${{ github.ref }}"
+            echo "Skip publish the image"
+            export PUBLISH=false
+          fi
+
+          IMAGE_NAME=coordinator-with-clp-connector-runtime \
+          REG_ORG=ghcr.io/${{github.actor}}/presto \
+          PUBLISH=$PUBLISH \
+          ./build.sh 0.293
+
+          rm presto-server-0.293.tar.gz
+          rm presto-cli-0.293-executable.jar
```

Comment:
Instead of relying on `build.sh`, can we instead do something like [this](https://github.com/kirkrodrigues/presto/blob/release-0.293-clp-connector/.github/workflows/maven-checks.yml#L64-L112)?

Similarly, I think in the native image build, it would be better to use the Dockerfiles directly rather than modifying the docker-compose file and then using docker compose.